### PR TITLE
SharedTokenCacheCredential lazily loads the cache

### PR DESCRIPTION
**Pull request by chlowell**
When we try to load the persistent cache on Linux and PyGObject isn't available, `msal-extensions` logs an error message. The message is only relevant to someone who wants to use the encrypted shared cache. However, because `SharedTokenCacheCredential.__init__()` and `.supported()` attempt to load the cache, `DefaultAzureCredential()` does as well. Many users will therefore see an alarming irrelevant log message (see #12152). This PR limits the alarm by having `SharedTokenCacheCredential` attempt to load the cache only when `get_token` is called for the first time.

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py [hunk 1]
```diff
@@ -55,6 +55,9 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         if not scopes:
             raise ValueError("'get_token' requires at least one scope")
 
+        if not self._initialized:
+            self._initialize()
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py:59]: You have 
def _initialize(self):
        if self._initialized:
            return
seems like you dup the work

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py:59]: It's defense against someone calling `_initialize` twice. But that shouldn't happen and if it does, the consequence is recreating the client, which isn't destructive. I removed the double-check.

Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py:59]: To make sure we are on same page, what I meant was:

Keep 
```python
def _initialize(self):
        if self._initialized:
            return
```
But modify
```python
if not self._initialized:
            self._initialize()
```
to
```python
self._initialize()
```

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py:59]: Ah, I see. I implemented it this way to avoid an additional call.

Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py:59]: If I understand correctly, calling into _initialize() twice may cause problem. 

You have allow_unencrypted = self._client_kwargs.pop("allow_unencrypted_cache", False) in your code which makes second call may have different value.

I agree with your original statement:
It's defense against someone calling _initialize twice.

Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_credentials/shared_cache.py:59]: Good eye, and you've reignited my paranoia :suspect:

**Pull request merged**